source/server/active_raw_udp_listener_config.cc

```diff
@@ -0,0 +1,26 @@
+#include "server/active_raw_udp_listener_config.h"
+
+#include "server/connection_handler_impl.h"
+
+namespace Envoy {
+namespace Server {
+
+Network::ConnectionHandler::ActiveListenerPtr
+ActiveRawUdpListenerFactory::createActiveUdpListener(Network::ConnectionHandler& parent,
+                                                     Network::ListenerConfig& config) const {
+  return std::make_unique<ConnectionHandlerImpl::ActiveUdpListener>(
+      dynamic_cast<ConnectionHandlerImpl&>(parent), config);
```

Comment:
This dynamic cast here is not great. Why do we need this? This is a good indication we have an abstraction/interface problem and we should take a look to see if we can clean this up somehow.

Reply:
ActiveUdpListenerImplBase actually access members of the impl class, i.e. stats_, rather than ConnectionHandler interface because it is a member class of the impl.

Reply:
Right but this again means we don't have the right interface/abstraction layers here. Can we figure out a way to untangle this?

Reply:
Most member access to ConnectionHandlerImpl is in ActiveTcpListener actually, in ActiveUdpListener, we only need to access dispatcher as of now. For quic, we probably will need to access dispatcher, logger, num_connection counter in the future. This is far smaller set of private members access compare to ActiveTcpListener. Because we don't create ActiveTcpListener via factory, ActiveTcpListener instantiation can still take an impl object. For ActiveUdpListener and ActiveQuicListener, their factories can take the ConnectionHandler interface instead, along with dispatcher and logger which are not exposed by the interface.  As a result, ActiveListenerImplBase::parent_ which is a ConnectionHandlerImpl should be moved to its derived class ActiveTcpListener now. And factory-created ActiveListeners should hold a handle to ConnectionHandler instance rather than the impl. ActiveUdpListener doesn't use ConnectionHandler anywhere other than accessing dispatcher, so it doesn't need to hold a handle of that. 

When we add ActiveQuicListener which modifies num_connections_ counter, we will need to add setter to ConnectionHandler interface. And